source/slang/slang-ir-wrap-global-context.cpp

```diff
@@ -0,0 +1,286 @@
+                    if (auto name = findNameHint(globalParam))
+                        builder.addNameHintDecoration(globalVar, name);
+                    mapGlobalParamToGlobalVar[globalParam] = globalVar;
+                }
+            }
+
+            // For every entry point, we need to add a new parameter for each global parameter.
+            for (auto entryPoint : entryPoints)
+            {
+                auto firstBlock = entryPoint->getFirstBlock();
+                auto paramInsertPoint = firstBlock->getFirstInst();
+                struct ParamInfo
+                {
+                    IRInst* newParam;
+                    IRInst* globalVar;
+                };
+                List<ParamInfo> newParams;
+                for (auto globalParam : mapGlobalParamToGlobalVar)
+                {
+                    auto newParam = builder.createParam(globalParam.first->getFullType());
+                    newParam->insertBefore(paramInsertPoint);
+
+                    newParams.add({newParam, globalParam.second});
+                }
+
+                // Insert assignments to the global variables at the start of the entry point.
+                builder.setInsertBefore(firstBlock->getFirstOrdinaryInst());
+                for (auto& paramInfo : newParams)
+                {
+                    auto globalVar = paramInfo.globalVar;
+                    auto newParam = paramInfo.newParam;
+                    builder.emitStore(globalVar, newParam);
+                }
+            }
+
+            // Replace all uses of global parameters with a load from the global variable.
+            for (auto globalParam : mapGlobalParamToGlobalVar)
+            {
+                auto globalVar = globalParam.second;
+                traverseUses(globalParam.first, [&](IRUse* use)
+                    {
+                        auto user = use->getUser();
+                        builder.setInsertBefore(user);
+                        auto load = builder.emitLoad(globalParam.first->getFullType(), globalVar);
+                        builder.replaceOperand(use, load);
+                    });
+                globalParam.first->removeAndDeallocate();
+            }
+        }
+
+        void processModule(IRModule* module)
+        {
+            IRBuilder builder(module);
+            List<IRInst*> instsToRemove;
+
+            List<IRFunc*> functions;
+
+            // Collect all entry points and functions.
+            for (auto globalInst : module->getGlobalInsts())
+            {
+                if (globalInst->findDecoration<IREntryPointDecoration>())
+                    entryPoints.add(as<IRFunc>(globalInst));
+                if (auto func = as<IRFunc>(globalInst))
+                    functions.add(func);
+            }
+
+            // Before everything, we need to move all global parameters to the entry point parameters.
+            // For each global parameter, e.g. `uniform float4 g;`, with will replace it with a global
```

Comment:
nit: 
float4 g;`, we will replace...

Reply:
Yes, I got confused there to.
"with" should be "we".